certbot/certbot/plugins/dns_test_common.py

```diff
@@ -1,22 +1,48 @@
 """Base test class for DNS authenticators."""
+try:
+    from typing import Protocol
+except ImportError:
+    Protocol = object  # type: ignore
```

Comment:
nit: What do you think about structuring this slightly differently here and elsewhere this pattern is used? I was thinking:
```suggestion
import typing

if typing.TYPE_CHECKING:
    from typing import Protocol
else:
    Protocol = object  # type: ignore
```
My thought is that trying to type check our code without `Protocol` should be an error.

Reply:
Indeed, and it is more aligned with mypy recommendations on that matter.